extensions/kubernetes/deployment/src/main/java/org/apache/camel/quarkus/component/kubernetes/cluster/deployment/KubernetesClusterServiceProcessor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.kubernetes.cluster.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import org.apache.camel.quarkus.component.kubernetes.cluster.KubernetesClusterServiceConfig;
+import org.apache.camel.quarkus.component.kubernetes.cluster.KubernetesClusterServiceRecorder;
+import org.apache.camel.quarkus.core.deployment.spi.RuntimeCamelContextCustomizerBuildItem;
+
+class KubernetesClusterServiceProcessor {
+
+    @Record(value = ExecutionTime.RUNTIME_INIT)
```

Comment:
We could maybe move this to `STATIC_INIT`.

Instead of a `CamelContextCustomizer`, we could produce a registry bean which `camel-main` will auto discover and add as a service on the `CamelContext`.

WDYT?

Reply:
It's more about moving the config phase to `STATIC_INIT`.

I forget if that can be achieved with `CamelContextCustomizer`. But by producing a registry bean, `camel-main` can automatically discover it and take care of registering the `ClusterService`.

Reply:
I see exposing the KubernetesClusterService as risky and moving to static init great if possible.
I doubt the CamelContextCustomizer could do that. I will experiment to feel what is the right balance.

Reply:
After deeper investigation, I have not found any way to execute `CamelContextCustomizer` at `STATIC_INIT`.
Plus, I have not found any simple way for users to intercept/break the cluster service from the registry.
So, registering at `STATIC_INIT` seems to be the best option here.
This is done in the last commit.

Reply:
Ok, I'm gonna polish that accordingly then.